lib/Sema/TypeCheckType.cpp

```diff
@@ -305,9 +305,11 @@ Type TypeChecker::resolveTypeInContext(
       continue;
 
     // Search the type of this context and its supertypes.
-    for (auto fromType = resolver->resolveTypeOfContext(parentDC);
-         fromType;
-         fromType = getSuperClassOf(fromType)) {
+    auto fromType = resolver->resolveTypeOfContext(parentDC);
+    for (;;) {
```

Comment:
Oh, I see. You could declare `superClassOfFromType` outside the loop, I guess, and use `fromType = superClassOfFromType` as your 3rd clause.